pyuvsim/uvsim.py

```diff
@@ -522,7 +530,8 @@ def uvdata_to_task_iter(task_ids, input_uv, catalog, beam_list, beam_dict, Nsky_
     # some additional overhead for numpy arrays as well
     # usually seeing [0.00001, 0.00003] MiB / task memory required
     # based on the reference simulations.
-    order = np.lexsort((_bls[task_slice], _freqs[task_slice], _times[task_slice])).flat
+    # Flatten inside the loop so that we always have the same thing in each loop.
```

Comment:
This comment is perhaps confusing.  Without knowing that this was the source of a bug, it seems weird to have something basically saying "Don't flatten here, flatten later".  Do we want to instead have a comment near line 564?

Reply:
I think I'll second this. Maybe just commend on like 564 to use flat so we have a 1-dimensional iterator. That was really the point of it to begin with since the iterable is originally an (nblts, freqs) array.

Reply:
will do!